tests/apichecks/crds_test.go

```diff
@@ -242,3 +242,33 @@ func visitProps(props *apiextensions.JSONSchemaProps, fieldPath string, callback
 		klog.Fatalf("unhandled props.Type %q in %+v", props.Type, props)
 	}
 }
+
+func TestCRDCamelCase(t *testing.T) {
+	crds, err := crdloader.LoadAllCRDs()
+	if err != nil {
+		t.Fatalf("error loading crds: %v", err)
+	}
+	var errs []string
+	for _, crd := range crds {
+		for _, version := range crd.Spec.Versions {
+			visitCRDVersion(version, func(field *CRDField) {
+				fieldPath := field.FieldPath
+				first := func() int32 {
+					tokens := strings.Split(fieldPath, ".")
+					// Only check the last token to avoid duplication.
+					for _, first := range tokens[len(tokens)-1] {
+						return first
+					}
+					return 0
+				}()
+				if unicode.IsLetter(first) && unicode.IsUpper(first) {
+					errs = append(errs, fmt.Sprintf("[refs] crd=%s version=%v: field %q should use camel case", crd.Name, version.Name, field.FieldPath))
```

Comment:
Nit: I was starting to put the "rule name" in square brackets, so maybe `[jsonNaming]`